### ISSUE TITLE
Unable to use chart.marginRight

### ISSUE THREAD

**Issue by ndrwkim** [highcharts-for-python/highcharts-core#124]
Trying to use [chart.marginRight](https://api.highcharts.com/highcharts/chart.marginRight) but it results in an error:

```
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_maps/chart.py", line 1006, in from_options
    options = validate_types(options, HighchartsOptions)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/decorators.py", line 102, in validate_types
    value = primary_type.from_dict(value)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/metaclasses.py", line 370, in from_dict
    return cls(**kwargs)
           ^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/options/__init__.py", line 714, in __init__
    super().__init__(**kwargs)
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/options/__init__.py", line 76, in __init__
    self.chart = kwargs.get('chart', None)
    ^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/decorators.py", line 260, in func_wrapper
    value = validate_types(value,
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/decorators.py", line 102, in validate_types
    value = primary_type.from_dict(value)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/metaclasses.py", line 370, in from_dict
    return cls(**kwargs)
           ^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/options/chart/__init__.py", line 168, in __init__
    self.margin = kwargs.get('margin', None)
    ^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/options/chart/__init__.py", line 595, in margin
    self.margin_top = value[0]
    ^^^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_core/options/chart/__init__.py", line 683, in margin_top
    self._margin_top = validators.numeric(value, allow_empty = True)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/validator_collection/_decorators.py", line 51, in func_wrapper
    return func(*args, **updated_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/validator_collection/validators.py", line 1433, in numeric
    raise errors.CannotCoerceError(
validator_collection.errors.CannotCoerceError: value (null) cannot be coerced to a numeric form
```

I'm trying to use it in a stacked bar template file. Example:
```
{
    chart: {
        type: "bar",
        marginRight: 100,
        scrollablePlotArea: {
            minHeight: 1000,
            opacity: 1
        }
    },
    ...
}
```

P.S. chart.spacingRight doesn't work either.

**Comment by hcpchris** [comment]
Hi @ndrwkim : Sorry you're running into this issue! I'll look into the issue shortly, but at first glance it looks like a bug that should be fairly fast / easy to fix. However, just to rule out certain things, can you confirm which version of Highcharts Core for Python (or looking at your traceback results, Highcharts Maps for Python) you are using?

**Comment by ndrwkim** [comment]
highcharts-core==1.3.7
highcharts-maps==1.3.0

I'm not using the most recent versions because I haven't figured out why they are breaking my implementation (despite the release notes stating they're backwards-compatible).

```
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_maps/chart.py", line 438, in add_series
    item_series = create_series_obj(item)
                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ak/git/alphabuyer/.env/lib/python3.11/site-packages/highcharts_maps/options/series/series_generator.py", line 206, in create_series_obj
    return instance
           ^^^^^^^^
UnboundLocalError: cannot access local variable 'instance' where it is not associated with a value
```
where `item` is a `BarSeries` instance.

The problem seems to be that `highcharts_maps.options.series.series_generator.create_series_obj` is checking `if isinstance(value, SeriesBase):` but this `SeriesBase` (`highcharts_maps.options.series.base.SeriesBase`) is different from the one that `BarSeries` inherits from (`highcharts_core.options.series.base.SeriesBase`) even though I'm importing via `from highcharts_maps.options.series.bar import BarSeries`.

**Comment by hcpchris** [comment]
Thanks - well, first off let me file a new bug related to that issue where the new version is generating an error on `create_series_obj` for you in Highcharts Maps for Python. That's definitely a bug and should not be the case, so I've created a [new issue](https://github.com/highcharts-for-python/highcharts-maps/issues/36) that over in the Highcharts Maps for Python repo. 

In the meantime, I'll also take a look at what the issue is with the margin issues in Highcharts Core for Python. Should hopefully have a fix for it released later today.

**Comment by hcpchris** [comment]
Hi @ndrwkim : Can you share the code snippet that you are using to generate the `CannotCoerceError` you are reporting? I have tried to replicate your error using the ``chart`` JS literal that you provided, as well as also supplying an identical `dict`, and I can't seem to get that exception to generate. I'm hoping that you can share the code snippet that you are using to produce that exception to help recreate the issue.

**Comment by hcpchris** [comment]
Also, just as a heads up, I did release a new version of Highcharts Maps for Python (v.1.4.1) which resolves the backwards compatibility issue that you reported. So you should be able to upgrade to the latest version of Highcharts Maps for Python and your existing code should still work - if it doesn't, then that would be another bug (which I will patch).

**Comment by ndrwkim** [comment]
I don't get an error anymore but all the charts are loading blank. Having difficulty troubleshooting this one with no errors.

**Comment by ndrwkim** [comment]
> Hi @ndrwkim : Can you share the code snippet that you are using to generate the `CannotCoerceError` you are reporting? I have tried to replicate your error using the `chart` JS literal that you provided, as well as also supplying an identical `dict`, and I can't seem to get that exception to generate. I'm hoping that you can share the code snippet that you are using to produce that exception to help recreate the issue.

```
    def foo(self, extra_filters=None):
        chart_options = HighchartsOptions.from_js_literal(self.CONFIG_DIR / 'stacked-bar-template-01.js')
        chart_options.title = {'text': str(_('foo'))}

        data = self._foo_data(extra_filters)
        opt_dict = chart_options.to_dict()
        opt_dict['xAxis']['categories'] = data['categories']

        chart = Chart.from_options(opt_dict, chart_kwargs={'container': 'foo'})

        # Iterate over series data and add to the chart.
        series_data_list = [
            BarSeries(data=series_data['data'], name=series_data['name']) for series_data in data['series']
        ]
        chart.add_series(*series_data_list)

        return chart
```

**Comment by hcpchris** [comment]
Thanks, @ndrwkim : Let me experiment with your sample code to try and see why the chart is not generating / rendering. Furthermore, your reporting that with the upgraded version of Highcharts Maps for Python all of your charts are loading blank (but had been loading fine with v.1.3.0) is an interesting data point. Let me see given the code you shared whether I can recreate that behavior. 

It may also have to do with your `stacked-bar-temlate-01.js` content. Would you be willing to share that with me, just to help rule out any details on it? If you're not comfortable posting that in a public Github repo, you can also email it to support@highchartspython.com as well.

**Comment by ndrwkim** [comment]
`stacked-bar-template-01.js`:
```
{
    chart: {
        type: "bar",
        marginRight: 100,
        scrollablePlotArea: {
            minHeight: 1000,
            opacity: 1
        }
    },
    exporting: {
        buttons: {
            contextButton: {
                menuItems: ["viewFullscreen", "printChart"]
            }
        }
    },
    legend: {
        enabled: true
    },
    navigation: {
        menuItemStyle: {
            color: "#5f5e5e",
            fontFamily: "Roboto",
            fontSize: "12px"
        }
    },
    tooltip: {
        formatter: function () {
            return `${this.series.name}: $${this.y.toLocaleString()}`;
        },

        style: {
            color: "#5f5e5e",
            fontFamily: "Roboto",
            fontSize: "12px",
            fontWeight: "700"
        }
    },
    xAxis: {
        visible: true,
    },
    yAxis: {
        labels: {
            enabled: true
        },
    },
    plotOptions: {
        series: {
            stacking: "normal",
            dataLabels: {
                enabled: false
            }
        },
        bar: {
            dataSorting: {
                enabled: false
            }
        }
    }
}
```

**Comment by hcpchris** [comment]
Thanks! I'll see if I can recreate the behavior that you are seeing (or rather, the behavior you are *not* seeing, if the chart is coming up blank).

**Comment by hcpchris** [comment]
Hi @ndrwkim : So good news. I think I found the nature of the bug, though I must admit it's a bit of a weird one. Also, I am not able to generate the "blank chart" scenario that you are encountering, at least not yet. However, I was able to find and resolve the bug causing the exception on ``marginRight`` and ``spacingRight``.

In this case, it's a bit of a weird edge case that I was able to uncover thanks to the sample code which you shared. Specifically, the error would occur when:

1. A ``ChartOptions`` instance that only had some margins set (e.g. only ``.margin_right``) was serialized using ``.to_dict()``. This had the effect of producing a serialised property like ``margin: ['null', 100, 'null', 'null']`` (which is correct).
2. A new ``ChartOptions`` instance was deserialized from the resulting ``dict`` using ``.from_dict()``. However, this specific situation would incorrectly handle a value of ``'null'`` in the ``dict``.

The same pattern was used for ``.margin`` and ``.spacing``, and I have corrected both in PR #125 . That will be released in **Highcharts Core for Python v.1.4.3** which should be released this evening. I'm currently waiting for the automated CI/CD pipelines to finish, at which point the PR will be merged and the new version released. This issue should auto-close when the version has been released. 

If you are still encountering issues after upgrading to the new version of Highcharts Core for Python - including the blank chart issue you mentioned, please either re-open this issue or open a new one: Happy to continue troubleshooting as needed.

**Comment by hcpchris** [comment]
Confirming the bugfix was released in **Highcharts Core for Python v.1.4.3**
